sprockets/mixins/http/__init__.py

```diff
@@ -6,50 +6,248 @@
+
+        :rtype: int
+
+        """
+        return self._responses[-1].code if self._responses else 599
+
+    @property
+    def duration(self):
+        """Return the calculated duration for the total amount of time
+        across all retries.
+
+        :rtype: float
+
+        """
+        return (self._finish or time.time()) - self._start
+
+    @property
+    def exceptions(self):
+        """Return the list of exceptions raised when making the request.
+
+        :rtype: list(Exception)
+
+        """
+        return self._exceptions
+
+    @property
+    def headers(self):
+        """Return the HTTP Response headers as a dict.
+
+        :rtype: dict
+
+        """
+        if not self._responses:
+            return None
+        return dict(self._responses[-1].headers)
+
+    @property
+    def history(self):
+        """Return all of the HTTP responses for the request.
+
+        :rtype: list(tornado.httpclient.HTTPResponse)
+
+        """
+        return self._responses
+
+    @property
+    def links(self):
+        """Return the parsed link header if it was set, returning a list of
+        the links as a dict.
+
+        :rtype: list(dict()) or None
+
+        """
+        if not self._responses:
+            return None
+        if 'Link' in self._responses[-1].headers:
+            links = []
+            for l in headers.parse_link(self._responses[-1].headers['Link']):
+                link = {'target': l.target}
+                link.update({k: v for (k, v) in l.parameters})
+                links.append(link)
+            return links
+
+    @property
+    def ok(self):
+        """Returns `True` if the response status code was between 200 and 399.
+        Returns `False` if no responses were received or the response status
+        code was >= 400.
+
+        :rtype bool
+
+        """
+        if not self._responses:
+            return False
+        return 200 <= self._responses[-1].code < 400
+
+    @property
+    def raw(self):
+        """Return the raw tornado HTTP Response object
+
+        :rtype: tornado.httpclient.HTTPResponse
+
+        """
+        if not self._responses:
+            return None
+        return self._responses[-1]
+
+    def _decode(self, value):
+        """Decode bytes to UTF-8 strings as a singe value, list, or dict.
+
+        :param mixed value: The value to decode
+        :rtype: mixed
+
+        """
+        if isinstance(value, list):
+            return [self._decode(v) for v in value]
+        elif isinstance(value, dict):
+            return {self._decode(k): self._decode(v)
+                    for k, v in value.items()}
+        elif isinstance(value, bytes):
+            return value.decode('utf-8')
+        return value
+
+    def _deserialize(self):
+        """Try and deserialize a response body based upon the specified
+        content type.
+
+        :rtype: mixed
+
+        """
+        if not self._responses or not self._responses[-1].body:
```

Comment:
nit - having a local variable for `self._responses[-1]` would be easier to read since it's used a bunch in this func.

Reply:
Yes, but it would create inefficiencies in the python stack with an extra op that is not required.